tasks/eva_3450/src/main/groovy/eva3450/CreateMergeCandidates.groovy

```diff
@@ -0,0 +1,128 @@
+package eva3450
+
+import org.slf4j.LoggerFactory
+import uk.ac.ebi.ampt2d.commons.accession.core.models.EventType
+import uk.ac.ebi.eva.accession.clustering.configuration.batch.io.RSMergeAndSplitCandidatesReaderConfiguration
+import uk.ac.ebi.eva.accession.core.EVAObjectModelUtils
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantInactiveEntity
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantOperationEntity
+import uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment
+
+import static org.springframework.data.mongodb.core.query.Criteria.where
+import static org.springframework.data.mongodb.core.query.Query.query
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getDbsnpSveClass
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getSveClass
+
+class CreateMergeCandidates {
+    static def logger = LoggerFactory.getLogger(CreateMergeCandidates.class)
+
+    String assembly
+    String mergeCandidatesFile
+    EVADatabaseEnvironment prodEnv
+    EVADatabaseEnvironment devEnv
+    int batchSize
+
+    CreateMergeCandidates() {}
+
+    CreateMergeCandidates(String assembly, String mergeCandidatesFile, EVADatabaseEnvironment prodEnv,
+                          EVADatabaseEnvironment devEnv, int batchSize) {
+        this.assembly = assembly
+        this.mergeCandidatesFile = mergeCandidatesFile
+        this.prodEnv = prodEnv
+        this.devEnv = devEnv
+        this.batchSize = batchSize
+    }
+
+
+    def process = {
+        def is = new File(mergeCandidatesFile).newInputStream()
+        logger.info("Create Merge candidate documents in batches of ${batchSize}")
+        int totalInsertedCount = 0
+        int lineRead = 0
+        def mergedCandidateOperations = []
+        ArrayList<String> batchedLines = new ArrayList<String>()
+        is.eachLine{
+            lineRead = lineRead + 1
+            batchedLines.add(it)
+            if (batchedLines.size() == batchSize){
+                def mergeCandidateOperations = createBatchMergeCandidates(batchedLines)
+                totalInsertedCount = insertOperations(mergeCandidateOperations, totalInsertedCount, lineRead)
+                batchedLines = new ArrayList()
+            }
+        }
+        if (batchedLines.size() > 0){
+            def mergeCandidateOperations = createBatchMergeCandidates(batchedLines)
+            totalInsertedCount = insertOperations(mergeCandidateOperations, totalInsertedCount, lineRead)
+        }
+        is.close()
+
+    }
+
+    def createBatchMergeCandidates = { List<String> hashAndRsids ->
+        Map<String, Long[]> hashToRsids = hashAndRsids.collectEntries {
+            String[] sp_line =  it.split("\\t")
+            def hash = sp_line.first()
+            def rsids = sp_line.tail().collect({it.toLong()})
+            return [ (hash): rsids]
+        }
+        def rsids = hashToRsids.values().toList().flatten()
```

Comment:
`flatten` is not needed, and actually I don't think it needs to be converted to a list either - I think the query just needs a `Collection` which `values()` will give you.

Reply:
I might be wrong but `hashToRsids.values()` give a `Collection` of `Long[]`. To get to a list of `Long`, I need to flatten it and I had to convert it to a List because `flatten()` didn't exist on `Collection`

Reply:
You're right, I forgot there was the extra `Long[]` level